packages/hooks/src/useStoreWalletAccountsList.ts

```diff
@@ -0,0 +1,123 @@
+    Demo: {
+        dark: 'IcWalletDerivDemoDark',
+        light: 'IcWalletDerivDemoLight',
+    },
+    USD: {
+        dark: 'IcWalletCurrencyUsd',
+        light: 'IcWalletCurrencyUsd',
+    },
+    EUR: {
+        dark: 'IcWalletCurrencyEur',
+        light: 'IcWalletCurrencyEur',
+    },
+    AUD: {
+        dark: 'IcWalletCurrencyAud',
+        light: 'IcWalletCurrencyAud',
+    },
+    GBP: {
+        dark: 'IcWalletCurrencyGbp',
+        light: 'IcWalletCurrencyGbp',
+    },
+    BTC: {
+        dark: 'IcWalletBitcoinDark',
+        light: 'IcWalletBitcoinLight',
+    },
+    ETH: {
+        dark: 'IcWalletEthereumDark',
+        light: 'IcWalletEthereumLight',
+    },
+    USDT: {
+        dark: 'IcWalletTetherDark',
+        light: 'IcWalletTetherLight',
+    },
+    eUSDT: {
+        dark: 'IcWalletTetherDark',
+        light: 'IcWalletTetherLight',
+    },
+    tUSDT: {
+        dark: 'IcWalletTetherDark',
+        light: 'IcWalletTetherLight',
+    },
+    UST: {
+        dark: 'IcWalletTetherDark',
+        light: 'IcWalletTetherLight',
+    },
+    LTC: {
+        dark: 'IcWalletLiteCoinDark',
+        light: 'IcWalletLiteCoinLight',
+    },
+    USDC: {
+        dark: 'IcWalletUsdCoinDark',
+        light: 'IcWalletUsdCoinLight',
+    },
+};
+
+/** A custom hook to get the list of wallets for the current user. */
+const useStoreWalletAccountsList = () => {
+    const { client } = useStore();
+    const { accounts, is_crypto } = client;
+
+    // Add additional information to each wallet.
+    const wallets = useMemo(() => {
+        return Object.keys(accounts)
+            ?.filter(id => accounts?.[id].account_category === 'wallet')
+            ?.map(id => {
+                const wallet = accounts?.[id];
+
+                const loginid = id;
+                const currency = wallet.currency;
+                const is_disabled = Boolean(wallet.is_disabled);
+                const is_virtual = Boolean(wallet.is_virtual);
+
+                const fiat_or_crypto = is_crypto(currency) ? 'crypto' : 'fiat';
+                const icon_type = is_virtual ? 'demo' : fiat_or_crypto;
+                const landing_company_name = wallet.landing_company_name?.replace('maltainvest', 'malta');
+                const is_malta_wallet = landing_company_name === 'malta';
+                const dtrade_loginid = wallet?.linked_to?.find(account => account?.platform === 'dtrade')?.loginid;
+                const dtrade_balance = accounts?.[dtrade_loginid ?? '']?.balance;
+
+                const wallet_currency_type = is_virtual ? 'Demo' : currency || '';
+                const icons = currency_to_icon_mapper[wallet_currency_type];
+
+                const gradients = {
+                    /** The gradient class name for the wallet header background. */
+                    header: {
+                        dark: `wallet-header__${wallet_currency_type.toLowerCase()}-bg--dark`,
```

Comment:
will this hook still work with another component? or if I change UI, will this still work?

basically, loose copuling is preferred over strong coupling whenever makes sense, also, typically hooks handle data structures,

Reply:
@wojciech-deriv honestly, doesn't understand what you mean. It's what  we discussed with you in another PR about class names. We have these classes described in core package because we have 4 types of wallets gradients for every currency (header-bg, header-bg-dark, card-bg, card-bg-dark) and we are using the same classes for all wallets related gradients. If the design changes we just need to change class description and doesn't need to change class names. About the rest hook data, all this data we need to show the switcher correctly (icon, class, dtrader_loginid and balance, etc.)

Reply:
I'm promoting good practices - in this case, https://en.wikipedia.org/wiki/Loose_coupling

Hook should not be aware that specific component exist. Should be general. This hook returns wallet account list and also header classes for a very specific component, meaning, that there is tight coupling. 

I'm happy to proceed and merge this (thus I approved PR), but wanted to avoid old patterns from legacy code to carry on to newer bits :)

Reply:
@wojciech-deriv Got it. Thanks for your opinion. I'll try to improve the code quality in the future and want to become a software architect and not just usual developer. Thank you again for your point